plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2722,7 +2722,9 @@ public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle
             metastore.truncateUnpartitionedTable(session, handle.getSchemaName(), handle.getTableName());
         }
         else {
-            for (HivePartition hivePartition : partitionManager.getOrLoadPartitions(metastore, handle)) {
+            Iterator<HivePartition> partitions = partitionManager.getPartitions(metastore, handle);
```

Comment:
Do we skip the partition check during `DELETE`

Reply:
Yeah, I don't think there's a real reason to limit the number of partitions for delete. Also the property says "scan".